### PR TITLE
Android Tracker 1.2.0

### PR DESCRIPTION
**Pull request by oguzhanunlu**

### CODE REVIEW

**Comment by yalisassoon** [comment]
Have also changed the PR to be into next.snowplowanalytics.com rather than master.

**Comment by oguzhanunlu** [comment]
Thanks @yalisassoon !

File: _posts/2019-07-03-snowplow-android-tracker-1.2.0-released/index.md [hunk 1]
```diff
@@ -9,6 +9,8 @@ permalink: /blog/2019/07/03/snowplow-android-tracker-1.2.0-released-with-global-
 discourse: true
 ---
 
+**IMPORTANT NOTE: 1.2.0 has a critical bug which causes duplication of natively supported contexts. 1.2.1 release is published to patch this bug. Thanks for the understanding.**
```
Comment (chuwy) [inline _posts/2019-07-03-snowplow-android-tracker-1.2.0-released/index.md]: IMO "Thanks for the understanding" feels a bit bureaucratic.

File: _posts/2019-07-03-snowplow-android-tracker-1.2.0-released/index.md [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+2. It makes the tracking less error-prone - it is not significantly less likely that an event will accidentally be sent without a required context attached.
+3. That in turn makes it easier for analysts and developers consuming the data, because they can be confident to expect certain events from particular platforms to always have the required contexts attached.
+
+Global contexts were introduced in the Javascript tracker in [version 2.10.0][js-tracker-2.10.0-blog-post]. This release of the Android tracker adds Global contexts to Android. We plan to release similar functionality for iOS apps in a forthcoming release of our Objective-C Tracker.
+
+<h3>1.3 Getting started with global contexts</h3>
+
+#### 1.3.1 Context Primitives
+
+Context primitives are simplest forms to represent global contexts and there are 2 forms: Self describing JSON & Context Generator.
```
Comment (LyubaGolovina) [inline _posts/2019-07-03-snowplow-android-tracker-1.2.0-released/index.md]:
```suggestion
Context primitives are the simplest forms to represent global contexts and there are 2 forms: Self describing JSON & Context Generator.
```

File: _posts/2019-07-03-snowplow-android-tracker-1.2.0-released/index.md [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+##### 1.3.2.2 RuleSet Provider
+
+A Ruleset Provider is used when you want to attach a global context to certain events based on the schema URI.
+
+Here is the constructor signature: `RuleSetProvider(String tag, RuleSet ruleSet, ContextPrimitive contextPrimitive)`
+- `tag` : the identifier of this context
+- `ruleSet` : A `RuleSet` object
+- `contextPrimitive` : the context primitive to be attached in case event schema is allowed per `ruleSet`
+
+`RuleSet` stores allowed and rejected URIs. Here is the signature of its' constructor: `RuleSet(String accept, String reject)` . Plural version is also available.
```
Comment (LyubaGolovina) [inline _posts/2019-07-03-snowplow-android-tracker-1.2.0-released/index.md]:
```suggestion
`RuleSet` stores allowed and rejected URIs. Here is the signature of its constructor: `RuleSet(String accept, String reject)` . Plural version is also available.
```

File: _posts/2019-07-03-snowplow-android-tracker-1.2.0-released/index.md [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+category: Releases
+permalink: /blog/2019/07/03/snowplow-android-tracker-1.2.0-released-with-global-contexts/
+discourse: true
+---
+
+**IMPORTANT NOTE: 1.2.0 has a critical bug which causes duplication of natively supported contexts. 1.2.1 release is published to patch this bug. Thanks for the understanding.**
+
+We are pleased to announce the `1.2.0` release of the [Snowplow Android Tracker][repo].
+
+[1.2.0][release-notes] introduces support for global contexts: introducing a set of new functions enabling users to send custom contexts with every event or a subset of events based on user provided criteria.
```
Comment (LyubaGolovina) [inline _posts/2019-07-03-snowplow-android-tracker-1.2.0-released/index.md]:
```suggestion
[1.2.0][release-notes] introduces support for global contexts: a set of new functions enabling users to send custom contexts with every event or a subset of events based on user provided criteria.
```

File: _posts/2019-07-03-snowplow-android-tracker-1.2.0-released/index.md [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+
+With Snowplow:
+
+* There are multiple contexts that are tracked out-of-the-box.
+* Users can define their own contexts. There is no limit to the number of contexts that can be defined, or the number that are attached to every event. This means Snowplow users can track very rich data.
+* Context data is loaded in a consistent way in the data warehouse, making it easy to query. So all the user-level data will be in the same user table, regardless of which event is fired.
+
+<h3>1.2 Introducing "global contexts"</h3>
+
+Until recently devleopers integrating Snowplow tracking had to attach contexts to each event.
```
Comment (LyubaGolovina) [inline _posts/2019-07-03-snowplow-android-tracker-1.2.0-released/index.md]:
```suggestion
Until recently developers integrating Snowplow tracking had to attach contexts to each event.
```

**Pull request merged**